packages/reassure-compare/src/utils/validate.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from 'zod';
+import { hasDuplicateValues } from './array';
+
+const metadataSchema = z.object({
+  metadata: z.object({
+    branch: z.string(),
+    commitHash: z.string(),
```

Comment:
As these are optional seems like we need to add `.optional()` chained method: https://github.com/colinhacks/zod#optionals

```suggestion
    branch: z.string().optional(),
    commitHash: z.string().optional(),
```